tests/std/tests/P0896R4_ranges_alg_copy_backward/test.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+// Validate that copy_backward_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::copy_backward_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::copy_backward(borrowed<false>{}, static_cast<int*>(nullptr))),
+    ranges::copy_backward_result<ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::copy_backward(borrowed<true>{}, static_cast<int*>(nullptr))),
+    ranges::copy_backward_result<int*, int*>>);
+
+struct instantiator {
+    static constexpr int input[] = {13, 42, 1729};
+
+    template <class Bidi1, class Bidi2>
+    static constexpr void call() {
+        // For the second range, we need an iterator to the end; it's expedient to simply ignore ranges with differing
+        // iterator and sentinel types (i.e., ranges that don't satisfy common_range).
+        if constexpr (ranges::common_range<Bidi2>) {
+            using ranges::copy_backward, ranges::copy_backward_result, ranges::iterator_t;
+            const Bidi1 wrapped_input{input};
+
+            { // Validate range overload
+                int output[] = {-2, -2, -2};
+                Bidi2 wrapped_output{output};
+                auto result = copy_backward(wrapped_input, wrapped_output.end());
+                STATIC_ASSERT(same_as<decltype(result), copy_backward_result<iterator_t<Bidi1>, iterator_t<Bidi2>>>);
+                assert(result.in == wrapped_input.end());
+                assert(result.out == wrapped_output.begin());

```

Comment:
Dont we need `peek()` here as `result.out` is not a sentinel?

Reply:
We only need to break the abstraction with `peek` when we have an input or output iterator and no sentinel to compare it with. The most common examples being a lone output iterator or an input iterator that an algorithm advances only partway through its range (so we have _a_ sentinel, but not one in the correct position.)